streams/src/test/java/org/apache/kafka/streams/kstream/TimeWindowsTest.java

```diff
@@ -19,50 +19,46 @@
 import org.apache.kafka.streams.kstream.internals.TimeWindow;
 import org.junit.Test;
 
-import java.time.Duration;
 import java.util.Map;
 
 import static java.time.Duration.ofMillis;
 import static org.apache.kafka.streams.EqualityCheck.verifyEquality;
 import static org.apache.kafka.streams.EqualityCheck.verifyInEquality;
-import static org.apache.kafka.streams.kstream.Windows.DEPRECATED_DEFAULT_24_HR_GRACE_PERIOD;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotEquals;
 import static org.junit.Assert.assertThrows;
 import static org.junit.Assert.fail;
 
-@SuppressWarnings("deprecation")
 public class TimeWindowsTest {
 
     private static final long ANY_SIZE = 123L;
     private static final long ANY_GRACE = 1024L;
 
     @Test
     public void shouldSetWindowSize() {
-        assertEquals(ANY_SIZE, TimeWindows.of(ofMillis(ANY_SIZE)).sizeMs);
```

Comment:
Let's leave this one in here, as long as we still support a deprecated API we should continue to test that it works. We just shouldn't use deprecated APIs to test other functionality unrelated to the API itself (eg that an exception is thrown for window size of 0).

Then we can restrict the scope of the deprecation warning to just this one test, rather than the entire class

Reply:
Updated in latest commit